go/vt/vtadmin/api_test.go

```diff
@@ -4594,9 +4600,16 @@ func TestGetWorkflows(t *testing.T) {
 	}
 }
 
+func loadTabletEnvFlags() {
+	_flag.ParseFlagsForTest()
+	fs := pflag.NewFlagSet("TestFlags", pflag.ContinueOnError)
+	tabletenv.RegisterTabletEnvFlags(fs)
```

Comment:
i'm very confident that vtadmin doesn't use these flags, do the tests fail without this change?

Reply:
Only this flag is needed: `--queryserver-config-pool-size`. Otherwise we get the error pasted below. Should I separately define that for this test for better clarity or leave it as is?
```
--- FAIL: TestVTExplain (0.03s)
    --- FAIL: TestVTExplain/authorized_actor (0.01s)
panic: invalid/out of range capacity [recovered]
	panic: invalid/out of range capacity

goroutine 171 [running]:
testing.tRunner.func1.2({0x2138e60, 0xc001da1090})
	/usr/local/go/src/testing/testing.go:1389 +0x24e
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1392 +0x39f
panic({0x2138e60, 0xc001da1090})
	/usr/local/go/src/runtime/panic.go:838 +0x207
vitess.io/vitess/go/pools.NewResourcePool(0xc001689c00?, 0x0?, 0x127e38e?, 0x3c2b040?, 0xc0016c4e10?, 0xc82485?, 0x2ce622a?)
	/home/rohit/vitess-ps/go/pools/resource_pool.go:127 +0x35a
vitess.io/vitess/go/vt/vttablet/tabletserver/connpool.(*Pool).Open(0xc001689c00, {0xc0001a2070?}, {0x3c2b040?}, {0xc0016c4ea0?})
	/home/rohit/vitess-ps/go/vt/vttablet/tabletserver/connpool/pool.go:126 +0x294
vitess.io/vitess/go/vt/vttablet/tabletserver.(*QueryEngine).Open(0xc0016dc360)
	/home/rohit/vitess-ps/go/vt/vttablet/tabletserver/query_engine.go:249 +0x29a
```

Reply:
so I think the problem is that `currentConfig` is no longer getting the values on `defaultConfig` copied in; i think for this code (not just this flag, more in general) to remain correct is for `currentConfig` to _start_ as exact copy of `defaultConfig`, instead of relying on the side effect of `init() { flag.Whatever }` from doing the copying over of the default values. 

let me know if that makes sense or not!

Reply:
You mean that we should be initialzing all the attributes of the `currentConfig` structure with the default values (same ones used in the flag definitions) as part of the `init()` in `tabletenv` ?

Reply:
yeah, we basically want `reflect.DeepEqual(defaultConfig, currentConfig)` to return true, i think

Reply:
Fixed in the last commit from last week.